### ISSUE TITLE
Feature request: warning about push_constant/uniform member's offset and alignment

### ISSUE THREAD

**Issue by technicallyagd** [KhronosGroup/glslang#1636]
This feature request actually comes in two parts.
### Warning about bug inducing member ordering
Consider this:
```glsl
layout(push_constant) uniform Pc {
  vec3 m1;
  vec4 m2;
  float m3;
  float m4;
} pc;
```
The offset of `m2` will be 16 instead of 12, and trying to access any following members will give you incorrect values. Even though this behavior follows the rules specified in section `14.5.4` of Vulkan spec, it is not immediately obvious to newcomers like myself that the bug relates to these rules. It would be nice if some warning would be raised while compiling to notify the developer about this potential problem and how it can be fixed (in this example, swapping `m2` and `m3`'s order will do).

### More helpful error message for incorrect offset value in layout
Additionally, I actually started out explicitly specifying the offset for all the members like this:
```glsl
layout(push_constant) uniform Pc {
  layout(offset=0) vec3 m1;
  layout(offset=12) vec4 m2;
  layout(offset=28) float m3;
  layout(offset=32) float m4;
} pc;
```
This is clearly incorrect now I know about the rules, but the error message given by the compiler is just `'offset' : must be a multiple of the member's alignment`, which isn't really all that helpful. Would it be possible to state which member is wrong and what the correct offset should be in the error message?

**Comment by johnkslang** [comment]
Do you do any reflection with glslang or other tools, to inspect in general what the interfaces of your shader are?

I am generally in favor of making things easier, but wonder if in this case it should come from reflection.  Warnings should be for something potentially wrong, but perhaps we could add "Notes" or so, on request.

> Would it be possible to state which member is wrong and what the correct offset should be in the error message?

This can likely be improved; I'll have a look.  There might not be "the correct offset", but maybe a default offset.

**Comment by technicallyagd** [comment]
> Do you do any reflection with glslang or other tools, to inspect in general what the interfaces of your shader are?

Not really, I am really new to shader programming, I just learned how to read the disassembly code while debugging the bug mentioned in the issue. Does the reflection API contain more information than those you get from the disassembly code?
> I am generally in favor of making things easier, but wonder if in this case it should come from reflection. Warnings should be for something potentially wrong, but perhaps we could add "Notes" or so, on request.

I think any form of notice to the user will help. I am aware that my request may be unreasonable or hard to achieve, since I don't have any knowledge regarding the internals of glslang. I just wanted other newcomers to spend less time than I did on problems related to this. (In a way, this issue itself may be enough now that this issue is the first result from google if you search for `'offset' : must be a multiple of the member's alignment`)
